crypto3/libs/blueprint/include/nil/blueprint/zkevm_bbf/exp.hpp

```diff
@@ -280,31 +282,31 @@ namespace nil {
                     }
 
                     for(std::size_t i = 0; i < max_rows; i++){
-                        allocate(header_selector[i], 0, i);
+                        allocate(header_selector[i], 0, i + 1);
 
                         for(std::size_t j = 0; j < num_chunks; j++){
-                            allocate(base[i][j], j + 1, i);
-                            allocate(exponent[i][j], num_chunks + j + 1, i);
-                            allocate(exponentiation[i][j], 2*num_chunks + j + 1,i);
+                            allocate(base[i][j], j + 1, i + 1);
+                            allocate(exponent[i][j], num_chunks + j + 1, i + 1);
+                            allocate(exponentiation[i][j], 2*num_chunks + j + 1,i + 1);
                         }
-                        allocate(is_last[i], 3*num_chunks + 1, i);
+                        allocate(is_last[i], 3*num_chunks + 1, i + 1);
                     }
 
                     for(std::size_t i=0; i < num_proving_blocks; i++){
                         for(std::size_t j = 0; j < 16; j++){
-                            allocate(r_chunks[i][j], 3*num_chunks + j + 2, 3*i);
-                            allocate(b_chunks[i][j], 3*num_chunks + j + 2, 3*i + 1);
-                            allocate(a_chunks[i][j], 3*num_chunks + j + 2, 3*i + 2);
+                            allocate(r_chunks[i][j], 3*num_chunks + j + 2, 3*i + 1);
+                            allocate(b_chunks[i][j], 3*num_chunks + j + 2, 3*i + 2);
+                            allocate(a_chunks[i][j], 3*num_chunks + j + 2, 3*i + 3);
```

Comment:
Let's avoid manual increment:
```suggestion
                            size_t row_to_allocate = 3*row_idx + 1;  // comment why this value
                            size_t chunk_to_allocate = 3*num_chunks + j + 2   // comment again
                            allocate(r_chunks[i][j], chunk_to_allocate, row_to_allocate++);
                            allocate(b_chunks[i][j], chunk_to_allocate, row_to_allocate++);
                            allocate(a_chunks[i][j], chunk_to_allocate, row_to_allocate++);
```

Reply:
It's a good principle, but I don't think, it's convenient here.
3*i, 3*i+1 e t.c. are used in multiple loops, and it's nice to understand where exactly similar row numbers are used.